### DISCUSSION TITLE
Custom fonts in TailwindCSS can't be loaded when hosted on Vercel

### DISCUSSION THREAD

**Discussion by juniorforlife** [remix-run/remix#4963] [Q&A]
I'm using TailwindCSS and I need to load some custom fonts. Below are my configs. It works fine on my local machine but not on the server

`remix.config.js`
```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  serverBuildTarget: "vercel",
  // When running locally in development mode, we use the built in remix
  // server. This does not understand the vercel lambda module format,
  // so we default back to the standard build output.
  server: process.env.NODE_ENV === "development" ? undefined : "./server.js",
  ignoredRouteFiles: ["**/.*"],
  // appDirectory: "app",
  assetsBuildDirectory: "public/build",
  // serverBuildPath: "api/index.js",
  publicPath: "/build",
};
```
Where I put my fonts
<img width="338" alt="Screen Shot 2022-12-26 at 23 04 39" src="https://user-images.githubusercontent.com/68360917/209566129-338b8c63-efc9-482d-a26e-50426eb2b138.png">

Custom font declaration in `styles/app.css` (TailwindCSS)
```

@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  @font-face {
    font-family: "Open Sans";
    src: url("/assets/fonts/OpenSans.ttf");
    font-weight: normal;
  }
  @font-face {
    font-family: "Open Sans";
    src: url("/assets/fonts/OpenSans-Bold.ttf");
    font-weight: 700;
  }
  @font-face {
    font-family: "Open Sans";
    src: url("/assets/fonts/OpenSans-SemiBold.ttf");
    font-weight: 700;
  }
}
```

Error message 
<img width="863" alt="Screen Shot 2022-12-26 at 23 21 05" src="https://user-images.githubusercontent.com/68360917/209567202-3ba653c2-4202-429e-9a5a-bdf60b5f505f.png">

**Comment by juniorforlife** [comment]
In case anyone would run into the same issue. Here are the steps you must do to get the local fonts work with TailwindCSS:
1. Put the fonts in `public/` folder
2. Define the font family in `styles/app.css` like this
```
@layer base {
  @font-face {
    font-family: "Open Sans";
    src: url("/fonts/OpenSans.ttf"); // no need for /public
  }
}
```
3. Add the fonts to `function links()` in `root.tsx`
```
export function links() {
  return [
    { rel: "stylesheet", href: styles }, // import styles from "/styles/app.css";
    {
      rel: "preload",
      as: "font",
      href: "/fonts/OpenSans.ttf", // must be the same with url above
      type: "font/ttf",
      crossOrigin: "anonymous",
    },
  ];
}
```

**Reply by flora-liu** [reply]
@juniorforlife Do you have an example code setup I could look at? I followed your instructions but it's still not working for me.

**Reply by juniorforlife** [reply]
what's the error you are seeing?
